### ISSUE TITLE
Intention of Transition: Removal of support for legacy MonoMac, XamMac platforms

### ISSUE THREAD

**Issue by hach-que** [Protobuild/Protobuild#164]
### What is an intention of transition?

When developing Protobuild, we strive to ensure that changes are forward compatible and don't break existing uses.  However, sometimes we need to change the default behaviour of Protobuild, either because of changing circumstances, new features or because the existing default behaviour is incorrect or faulty.

An intention of transition indicates that a default will be changing in a future of Protobuild, and outlines the reason for the change.

### What is changing?

With the announcement that the Xamarin APIs are now freely usable by everyone, with no license needed, we are dropping support for the legacy free APIs that were previously available for Mac OS.  With Xamarin being free, all users can upgrade to the latest version of the Mac APIs.

Along with this change, there'll be no more circumstances under which the `PLATFORM_MACOS_LEGACY` constant will be inserted.  Any code which is included based on the presence of this constant can be removed.  Any code which is excluded based on the presence of this constant can have the conditional check surrounding it removed.

### When is the change occurring?

This change will occur on or soon after the 25th May, 2016.

### What is affected?

On systems that do not have Xamarin.Mac installed (either missing entirely or an older version), projects will no longer build.  You will need to install Xamarin.Mac, which is freely available from [the Xamarin website](https://www.xamarin.com/download).

### What do I need to do to keep the existing behaviour?

The behaviour will only change when you start using a version of `Protobuild.exe` that includes the new behaviour.  If you keep your existing version of Protobuild, the behaviour will not change.

That is, in order to keep the current behaviour of generating projects for MonoMac and XamMac, you'll need to keep on a version built before May 25th, 2016.

**Comment by hach-que** [comment]
@tomspilman Heads up, this impacts you.  You will need to upgrade the version of Xamarin Studio installed on the Mac build server.

I want to drop this stuff because I'm continuously running into scenarios where Xamarin.Mac support in MonoGame keeps getting broken because people are running out of date versions of Xamarin Studio.  With it being free, there's no reason to be using the older versions.

**Comment by tomspilman** [comment]
Yeah...  want to see what @dellis1972 thinks, but I agree.  If Xamarin Studio is free to all and works better than MonoMac then I see no reason to keep supporting it.   Seems like less trouble for everyone.
